trin-execution/src/trie_walker.rs

```diff
@@ -34,6 +36,17 @@ pub struct TrieWalker {
 
 impl TrieWalker {
     pub fn new(root_hash: B256, nodes: BrownHashMap<B256, Vec<u8>>) -> Self {
+        // if the storage root is empty then there is no storage to gossip
+        if root_hash == keccak256([EMPTY_STRING_CODE]) && !nodes.is_empty() {
+            panic!("Root hash is empty but there are nodes to gossip. This should never happen.");
```

Comment:
If you are sure this shouldn't happen, leave it like this.
Otherwise change it to logging error and return empty map (same as below).

Reply:
This should be impossible if it did happen there is def a big mistake in the codebase